pkg/cli/cli_test.go

```diff
@@ -121,3 +121,30 @@ func Test_CLIRepoertErr(t *testing.T) {
 		t.Errorf("should return err status code: %d", exitStatus)
 	}
 }
+
+func Test_CLI_IgnoreBadPklFileWhenBinaryNotFound(t *testing.T) {
+	// Save the original function before mocking and restore it after the test
+	originalIsPklBinaryPresent := isPklBinaryPresent
+	isPklBinaryPresent = func() bool {
+		return false
+	}
+	defer func() { isPklBinaryPresent = originalIsPklBinaryPresent }()
```

Comment:
I'm afraid this could lead to data race, when other tests are running

Using a lock mechanism to solve this would be over complicated as it would be applied even with the CLI

Reply:
Tests within the same package run sequentially, so there shouldn't be a risk of a data race here. Lmk if that's not the case.

If we decide to make these parallel later, I think adding locking wouldn't be a bad idea. Can you explain what you mean by "it would be applied even with the CLI"?

Reply:
The locking I was suggesting would have been a mutex.Read/Write + defer unlock

to use it in the tests, I think it would require to use it in the code too, so when trying to check if isPklBinaryPresent here, when using it in the CLI, the lock would have to be present

The test are ran sequentially, but using a -race help you to spot such issues.